pkg/api/persistentvolume/util.go

```diff
@@ -111,6 +111,32 @@ func VisitPVSecretNames(pv *api.PersistentVolume, visitor Visitor) bool {
 		if source.StorageOS.SecretRef != nil && !visitor(source.StorageOS.SecretRef.Namespace, source.StorageOS.SecretRef.Name) {
 			return false
 		}
+	case source.CSI != nil:
+		if source.CSI.ControllerPublishSecretRef != nil {
+			if len(source.CSI.ControllerPublishSecretRef.Name) == 0 || len(source.CSI.ControllerPublishSecretRef.Namespace) == 0 {
```

Comment:
validation will prevent these from being empty... no need to recheck here (or in the other blocks)

Reply:
@liggitt could you please point me where the validation you mentioned should be added.

Reply:
https://github.com/kubernetes/kubernetes/blob/82eeda0885851eb7f56fd30422d4b0b0afe9a270/pkg/apis/core/validation/validation.go#L1432-L1456